File: helpdesk/src/Helpdesk.SPA/src/models/issueModel.js
import { DocumentModel } from './documentModel.js';
export class IssueModel {
  constructor(obj = {}) {
    this.id = obj.id || null;
    this.title = obj.title;
    this.phone = obj.phone;
    this.description = obj.description;
    this.htmlDescription = obj.htmlDescription;
    this.categoryId = obj.categoryId;
    this.category = obj.category || null;
    this.subcategoryId = obj.subcategoryId;
    this.subcategory = obj.subcategory || null;
    this.priorityId = obj.priorityId;
    this.priority = obj.priority || null;
    this.statusId = obj.statusId;
    this.status = obj.status || null;
    this.isEscalated = obj.isEscalated;
    this.isLevel3Support = obj.isLevel3Support;
    this.submitterUsername = obj.submitterUsername;
    this.createDate = obj.createDate;
    this.modifyDate = obj.modifyDate;
    this.comments = obj.comments || [];
    this.statusHistory = obj.statusHistory || [];
    this.institution = obj.institution;
    this.survey = obj.survey || null;
    this.requestForInformation = obj.requestForInformation || false;
    this.surveyObject = JSON.parse(this.survey);
    this.assignedToSysUserId = obj.assignedToSysUserId || null;
    this.assignedToSysUser = obj.assignedToSysUser;
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.events = this.comments.map(el => {return {createDate: el.createDate, type:'comment', uid: el.uid, element: el};})
      .concat(this.statusHistory.map(el => {return {createDate: el.createDate, type: 'status', uid: el.uid, element: el};}))
      .sort(function(a, b) {
        return (new Date(a.createDate) - new Date(b.createDate));
    });
  }
}

File: helpdesk/src/Helpdesk.Models/Issue/IssueModel.cs
using System;
using System.Collections.Generic;

namespace Helpdesk.Models.Issue
{
    public class IssueModel
    {
        public int? Id { get; set; } 
        public string Title { get; set; }
        public string Description { get; set; }
        public int CategoryId { get; set; }
        public int PriorityId { get; set; }
        public int? StatusId { get; set; }
        public int? SubcategoryId { get; set; }
        public bool IsEscalated { get; set; }
        public bool IsLevel3Support { get; set; }
        public int? AssignedToSysUserId { get; set; }
        public string AssignedToSysUser { get; set; }
        public DateTime? ResolveDate { get; set; }
        public string Phone { get; set; }
        public string Survey { get; set; }
        public IEnumerable<DocumentModel> Documents { get; set; }
        public bool RequestForInformation { get; set; }
    }
}
